### PR TITLE
K8SPSMDB-381: Get rid of using ENV vars

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-381](https://jira.percona.com/browse/K8SPSMDB-381) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| self-healing | failed |
| rs-shard-migration | passed |
| upgrade | passed |
| service-per-pod | passed |
| upgrade-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | failed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| self-healing | failed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | failed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-548-ed0d6008

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| arbiter | passed |
| scheduled-backup | passed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

File: pkg/psmdb/mongo/mongo.go [hunk 1]
```diff
@@ -81,6 +81,85 @@ func ReadConfig(ctx context.Context, client *mongo.Client) (RSConfig, error) {
...
+	}
+
+	if resp.OK != 1 {
+		return errors.Errorf("mongo says: %s", resp.Errmsg)
+	}
+
+	return nil
+}
+
+func CreateUser(ctx context.Context, client *mongo.Client, user, pwd string, roles []interface{}) error {
```
Comment (qJkee) [inline pkg/psmdb/mongo/mongo.go:120]: you can change `roles []interface{}` just to `roles ...interface{}` to get rid of []interface{}{"clusterMonitor"}

File: pkg/controller/perconaservermongodb/mgo.go [hunk 2]
```diff
@@ -350,37 +314,93 @@ func (r *ReconcilePerconaServerMongoDB) handleReplsetInit(m *api.PerconaServerMo
...
+	if err != nil {
+		return errors.Wrap(err, "failed to create clusterAdmin")
+	}
+
+	monitorUser, err := r.getInternalCredentials(cr, roleMonitorUser)
+	if err != nil {
+		return errors.Wrap(err, "failed to get cluster admin")
+	}
+
+	err = mongo.CreateUser(context.Background(), cli, monitorUser.Username, monitorUser.Password, []interface{}{"clusterMonitor"})
```
Comment (qJkee) [inline pkg/controller/perconaservermongodb/mgo.go:369]: can we use const `roleMonitorUser`

File: pkg/psmdb/mongo/mongo.go (hunk 1 above)
Comment (blez) [reply pkg/psmdb/mongo/mongo.go]: https://github.com/percona/percona-server-mongodb-operator/pull/548/commits/681651964d009c7ec4b0b95c1a4a7d902f4633ef

File: pkg/controller/perconaservermongodb/mgo.go (hunk 2 above)
Comment (blez) [reply pkg/controller/perconaservermongodb/mgo.go]: https://github.com/percona/percona-server-mongodb-operator/pull/548/commits/681651964d009c7ec4b0b95c1a4a7d902f4633ef

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| limits | passed |
| scheduled-backup | passed |
| scaling | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | failed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| upgrade | passed |
| liveness | passed |
| upgrade-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| operator-self-healing | passed |
| upgrade-sharded | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-548-68165196

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-548-68165196

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |

**Review by delgod — APPROVED** [review]

**Pull request merged**